aten/src/ATen/templates/RegisterDispatchKey.cpp

```diff
@@ -63,7 +72,22 @@ namespace {
 ${dispatch_anonymous_definitions}
 
 TORCH_LIBRARY_IMPL(aten, ${DispatchKey}, m) {
-  ${dispatch_registrations}
+  ${cpp_namespace}::Register${BackendName}${DispatchKey}Modules = [&]() {
+
+    ${dispatch_registrations}
+
+  };
+
+  // For external backends, the EAGER_REGISTRATION macro will be set to 1,
+  // causing the module registration function to be called at the same time
+  // it used to be called, in other words, at c++ static initialization time.
+  // For LTC backends (i.e. the lazy TorchScript backend) the EAGER_REGISTRATION
+  // macro will be set to 0 makes the c++ static initializer simply install
+  // a function pointer, and then only upon initializing the LTC backend
+  // (i.e. via a call to say init_ts_backend) is it finally invoked.
+  #if EAGER_REGISTRATION
+  ${cpp_namespace}::Register${BackendName}${DispatchKey}Modules();
```

Comment:
This might just be from a different PR, but I think we technically don't need `BackendName` here? (the function name will be unique without it). Plumbing a new `BackendName` argument through the codegen for the name seems like overkill to me, but maybe there are other places in the codegen where `BackendName` actually is required.

Reply:
@bdhirsh This is something I added to differentiate between backends. There is no guarantee as far as I am aware as to the order in which the symbols are statically initialized. This means that if we had just a `RegisterLazyModules` function being defined by both the TS backend and a custom backend, then there is no way of knowing who assigned the value of that function pointer last. To avoid ambiguity, I plumbed through the backend name to make things clear and unambiguous

Reply:
yeah good point, I can't think of an easy way to get around this. (cc @ezyang if you happen to have a different opinion. Context = this PR updates the codegen'd dispatcher registrations to be able to be delayed, and registered through a lambda instead of at load time).